src/inmanta/agent/executor.py

```diff
@@ -73,10 +116,10 @@ def generate_blueprint_hash(self) -> str:
     def __eq__(self, other: object) -> bool:
         if not isinstance(other, EnvBlueprint):
             return False
-        return self.generate_blueprint_hash() == other.generate_blueprint_hash()
+        return self.blueprint_hash() == other.blueprint_hash()
```

Comment:
MD5 is not guaranteed to be unique, is it? I get using it for `hash`, but `==` isn't completely accurate, or are hash collisions so rare that we don't care about it?